git-link.el

```diff
@@ -845,22 +860,25 @@ With a double prefix argument invert the value of
            (list remote nil nil)
          (list remote (car region) (cadr region))))))
 
-  (let (filename branch commit handler remote-info (remote-url (git-link--remote-url remote)))
+  (let ((remote-url (git-link--remote-url remote))
+        filename branch commit handler remote-info git-host web-host)
     (if (null remote-url)
         (message "Remote `%s' not found" remote)
 
       (setq remote-info (git-link--parse-remote remote-url)
+            git-host    (car remote-info)
             filename    (git-link--relative-filename)
             branch      (git-link--branch)
             commit      (git-link--commit)
-            handler     (git-link--handler git-link-remote-alist (car remote-info)))
+            handler     (git-link--handler git-link-remote-alist git-host)
+            web-host    (or (assoc-default git-host git-link-web-host-alist) git-host))
```

Comment:
We can use `assoc` since we're using `or` to provide a default; _or_, provide the default to `assoc-default`. I think the rest of code uses the `or` pattern so maybe that is a better choice?

Reply:
The problem with `assoc` is that it returns the pair `(key . value)`, so that we'd have `(cdr (assoc ...))`. No problem if that's what you prefer, but the extra `cdr` is why I didn't use it. As for `assoc-default`, unfortunately it respects the default value only for keys that are in the list but not associated with any value. For example:

```elisp
(assoc-default 'apple '((apple . red) pear) nil 'unknown) ;; -> red
(assoc-default 'pear '((apple . red) pear) nil 'unknown) ;; -> unknown
(assoc-default 'orange '((apple . red) pear) nil 'unknown) ;; -> nil, not unknown as we'd expect...
```

Reply:
Ah yes I see. Thanks for clarifying.

Looks good then, just one question: the other host-related alists do a regex match on the host, should we do that here for consistency?